lib/evaluation/handlers/subjectIsHolderEvaluationHandler.ts

```diff
@@ -54,25 +53,6 @@ export class SubjectIsHolderEvaluationHandler extends AbstractEvaluationHandler
     this.updatePresentationSubmission(pd);
   }
 
-  private generateIsHolderNotRequiredResults(pd: PresentationDefinition, vcs: VerifiableCredential[]) {
-    const indexes = [...Array(pd.input_descriptors.length).keys()].filter(
-      (s) => !this.isHolder.map((e) => e.path[2]).includes(s)
-    );
-    indexes.forEach((i) =>
-      vcs.forEach((_, index) =>
-        this.getResults().push(
-          this.createResult(
-            [],
-            `$.input_descriptors[${i}]`,
-            [`$[${index}]`, {}],
-            Status.INFO,
-            'is holder is not required'
-          )
-        )
-      )
-    );
-  }
-
```

Comment:
We should have restored the old/correct logic. And we should have only added the result objects which were missing.

Current solution if passes the test case will be accepted but I would be less confident of it's reliability.